### ISSUE TITLE
Rule proposal: forbid innerHTML prop on DOM element tags

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#712]
React will issue a warning in dev if you try to use pass `innerHTML` prop to a DOM element tag.

https://github.com/facebook/react/blob/85dcbf83/src/renderers/dom/shared/ReactDOMComponent.js#L182-L184

This should be easy enough to enforce for most cases with a linter rule.

Bad:

```jsx
<div innerHTML='HTML' />
```

Good:

```jsx
<div dangerouslySetInnerHTML={{ __html: 'HTML' }} />
```

**Comment by lencioni** [comment]
Name: `no-inner-html`?
